### DISCUSSION TITLE
How to do load and play custom sequences?

### DISCUSSION THREAD

**Discussion by 01GOD** [copych/AcidBox#31] [Q&A] [answered]
Seemed like a good idea to make a new discussion focused on this. How to do load and play custom drum sequences? For example, WHERE in that code is the instrument playing the pattern in sequencer_step() specified??? And where is it specified what drum samples are mapped to what numbers in a pattern when played from that sequencer_step() method? 

I modified acidbanger.ino to play a custom synth sequence, but it's still not playing a custom drum sequence. 

This is the format that plays a synth sequence (but how to get it to load and play as a drum (sampler) sequence?):
`
```
void OMNI_Make_A_Drum_Pattern() {
    // Declare and initialize a temporary newPattern instance
    OMNI_Pattern newPattern;

    // Initialize the newNotes array
    uint8_t newNotes[PatternLength] = {36,36,36,36,36,36,36,36,36,36,36,36,36,36,36,36,};

    // Assign the new pattern to the target pattern memory
    OMNI_memories[0].patterns[1] = newPattern;

    // Copy the notes array to the pattern's notes array using memcpy
    memcpy(OMNI_memories[0].patterns[1].notes, newNotes, sizeof(newNotes));

    // Set the accent and glide bitmasks
    OMNI_memories[0].patterns[0].accent = 0;
    OMNI_memories[0].patterns[0].glide = 0;
}
```

This is the main function I modified to get that to play as a synth sequence, although when tried adding a second pattern (as OMNI_memories[0].patterns[1] or OMNI_memories[1].patterns[0]) it didn't play a drum sequence.: 
`
```
void OMNI_sequencer_step(byte step) {
  //UNCOMMENT THE NEXT 2 LINES LATER: 
  OMNI_Make_A_Pattern();
  OMNI_Make_A_Drum_Pattern();
// #ifdef MIDI_RAMPS
//   do_midi_ramps();
// #endif
// #ifdef DEBUG_JUKEBOX_
//   DEBF("midi step %d\r\n", step);
// #endif
  // Play all notes in current step
  // for (int i = 0; i < NumInstruments; i++) {
    for (int i = 0; i < 2; i++) {
    OMNI_Pattern *pat = &OMNI_memories[OMNI_cur_memory].patterns[i];
    byte accent = (pat->accent >> step) & 1;
    byte glide = (pat->glide >> step) & 1;
    byte value = pat->notes[step];

    if (value > 0)
      instr_noteon(i, value, glide, accent);
    else
      instr_noteoff(i);
  }
  if (Break.after == bar_current && step == 0) {
    instr_noteon(NumInstruments - 1, 127, 0, 0);
    //    instr_noteon_raw(NumInstruments-1, CRASH_NOTE, 127, 0);
    // if (flip(30)) {
    //   //change drumkit
    //   current_drumkit = myRandom(((Drums.GetSamplesCount()-1)/12)) * 12 ;
      
// //#ifdef DEBUG_JUKEBOX
//       DEBF("Selected drumkit: %d\r\n" , current_drumkit);
// //#endif
//     }
// #ifdef DEBUG_JUKEBOX
//     DEBUG("CRASH!!!!!!!!!!!!!!!!!!!!!");
// #endif
// #ifdef MIDI_RAMPS
//     check_midi_ramps(true);
// #endif
  }
  // if (step % 4 == 0 || step == 1) {
  //   digitalWrite(LED_BUILTIN, HIGH);
  //   #ifdef LOLIN_RGB
  //     pixels.setPixelColor(0, pixels.Color(rand() % 32, rand() % 32, rand() % 32));
  //     pixels.show();
  //   #endif

// #ifdef DEBUG_JUKEBOX
//     DEBUG(step);
// #endif
  // }  else {
  //   digitalWrite(LED_BUILTIN, LOW);
  //   #ifdef LOLIN_RGB
  //     pixels.setPixelColor(0, pixels.Color(0,0,0));
  //     pixels.show();
  //   #endif
  // }
}
```

Thanks in advance for explaining how to load and play a SPECIFIC custom drum pattern! (with a SPECIFIC custom synth pattern)!

(This is maybe the first time I have ever seen a pattern randomizer made BEFORE a manual sequencer, and that has made it MUCH more challenging to figure out how to load a pattern into it and play it)

**Comment by copych** [comment, marked as answer]
For ones who find this. There's currently no ready-made interface for setting patterns manually. You can dig into Acidbanger.ino, or turn it off completely and run some sequencer elsewhere (on the same chip, on another chip, or use your DAW or whatever you like).
